pkg/domain/service/place/service.go

```diff
@@ -0,0 +1,136 @@
+package place
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+	placeEntity "wantum/pkg/domain/entity/place"
+	"wantum/pkg/domain/repository"
+	"wantum/pkg/domain/repository/place"
+	"wantum/pkg/werrors"
+
+	"google.golang.org/grpc/codes"
+)
+
+type Service interface {
+	Create(ctx context.Context, masterTx repository.MasterTx, name string) (*placeEntity.Entity, error)
+	Update(ctx context.Context, masterTx repository.MasterTx, placeID int, name string) (*placeEntity.Entity, error)
+
+	Delete(ctx context.Context, masterTx repository.MasterTx, placeID int) error
+	UpDeleteFlag(ctx context.Context, masterTx repository.MasterTx, placeID int) (*placeEntity.Entity, error)
+	DownDeleteFlag(ctx context.Context, masterTx repository.MasterTx, placeID int) (*placeEntity.Entity, error)
+
+	GetByID(ctx context.Context, masterTx repository.MasterTx, placeID int) (*placeEntity.Entity, error)
+	GetAll(ctx context.Context, masterTx repository.MasterTx) (placeEntity.EntitySlice, error)
+}
+
+type service struct {
+	placeRepository place.Repository
+}
+
+func New(repo place.Repository) Service {
+	return &service{
+		placeRepository: repo,
+	}
+}
+
+func (s *service) Create(ctx context.Context, masterTx repository.MasterTx, name string) (*placeEntity.Entity, error) {
+	createdAt := time.Now()
+	place := &placeEntity.Entity{
+		Name:      name,
+		CreatedAt: &createdAt,
+		UpdatedAt: &createdAt,
+	}
+	result, err := s.placeRepository.Insert(ctx, masterTx, place)
+	if err != nil {
+		return nil, werrors.Stack(err)
+	}
+	place.ID = result
```

Comment:
[MAY]これって、result自体にはIDが入っていないんじゃないか？？

```suggestion
	place.ID = result.LastInsertId()
```

Reply:
ここはresultに新しいデータのid(int)が入ってます！
ここちょっと命名的に紛らわしいな...

Reply:
ごめん！これサービス層の話だと思ってなくて、指摘してしまった、、、！！
resultだとSQLの結果みたいな印象を与えてしまうので、`placeData`とかに命名変更して欲しいかも！